File: app/src/main/java/com/engr195/spartansuperway/spartansuperway/MainFragment.java
package com.engr195.spartansuperway.spartansuperway;

import android.os.Bundle;
import android.support.v4.app.Fragment;

public class MainFragment extends Fragment {

    private static final String ARG_FIRST = "ARG_FIRST";
    private static final String ARG_LAST = "ARG_LAST";

    // TODO: Pass Firebase credentials here
    public static MainFragment newInstance(String first, String last) {
        MainFragment fragment = new MainFragment();
        Bundle args = new Bundle();
        args.putString(ARG_FIRST, first);
        args.putString(ARG_LAST, last);
        fragment.setArguments(args);
        return fragment;
    }
}

File: app/src/main/java/com/engr195/spartansuperway/spartansuperway/MainFragment.kt
package com.engr195.spartansuperway.spartansuperway

import android.content.Intent
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.fragment_main.*
import kotlinx.android.synthetic.main.dialog_payment.*

// TODO: Remove this fragment?
class MainFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? =
            inflater.inflate(R.layout.fragment_main, container, false)

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // TODO: Animate any changes when new views pop up.

        orderPodButton.setOnClickListener {
            // TODO: Set default visibility of payment form to 'View.GONE'
            paymentForm.visibility = View.VISIBLE
        }

        submitFormButton.setOnClickListener {
            // TODO: Set default visibility of QR image to 'View.GONE'
            paymentForm.visibility = View.GONE
            qrImage.visibility = View.VISIBLE
        }

        qrImage.setOnClickListener {
            val intent = Intent(activity, EtaActivity::class.java)
            startActivity(intent)
        }
    }
}